exampleCourse/questions/fibonacciEditorWithR/tests/tests.R

```diff
@@ -0,0 +1,47 @@
+# Unit tests with `testthat` use the behavior-driven development (BDD) format
+# instead of the traditional `test_that` format. This allows for two "contexts"
+# to be created.
+
+# In particular, we make the following assumptions about unit tests:
+
+# 1. `describe()` function indicates a single problem steps worth.
+# 2. `it()` function describes the specification being assessed.
+#
+# Avoid using multiple comparisons within the same `describe()`
+
+# Sample test
+describe("1", {                               # Specify points per check
+    it("Does `fib()` exist?", {               # Description of test
+        source("../student/student_answer.R") # Load student submission
+        expect_equal(class(fib), "function")  # Perform test
+    })
+})
+
+describe("1", {
+    it("Check fib(0)", {
+        source("../student/student_answer.R")
+        expect_equal(fib(0), 0)
+    })
+})
+
+describe("1", {
+    it("Check fib(1)", {
+        source("../student/student_answer.R")
+        expect_equal(fib(1), 1)
+    })
+})
+
+describe("3", {
+    it("Check fib(13)", {
+        source("../student/student_answer.R")
+        expect_equal(fib(13), 233)
+    })
+})
+
+describe("3", {
+    it("Check fib(n)", {
+        source("../student/student_answer.R")
+        student_ans = fib(15)
+        expect_equal(student_ans, 610)
```

Comment:
Best practice here would be to randomly generate a few different test values to check, to prevent students hard-coding answers.

Reply:
Agreed. I was trying to mimic here the `Python` implementation with unit tests.

Reply:
Yep, I see what you were doing here. I’d like to push these example questions to implement best practices, though, because I think people will copy them.

I think a good practical strategy here would be to add about 3 tests that each generate a random input. For example, replace the `fib(13)` and `fib(15)` checks with ones that choose a random value in ranges `[5,14]`, `[15,24]`, and a new one in `[25,34]`. You could add a comment saying that we choose random values so that students can’t hard-code solutions.